utils.py

```diff
@@ -0,0 +1,152 @@
+import os
+import shutil
+import sys
+import configparser
+import datetime
+from pathlib import Path
+from itertools import product
+from tqdm import tqdm
+from uuid import uuid4
+import pandas as pd
+from process_results import create_run_csv
+from imagenet.model_saver import save_model
+
+
+import smartsim
+from smartsim import Experiment, status
+from smartredis import Client
+
+
+from smartsim.log import get_logger, log_to_file
+logger = get_logger("Scaling Tests")
+
+def get_date():
+    date = str(datetime.datetime.now().strftime("%Y-%m-%d"))
+    return date
+
+def _check_model(device, force_rebuild=False):
+    if device.startswith("GPU") and (force_rebuild or not Path("./imagenet/resnet50.GPU.pt").exists()):
+        from torch.cuda import is_available
+        if not is_available():
+            message = "resnet50.GPU.pt model missing in ./imagenet directory. \n"
+            message += "Since CUDA is not available on this node, the model cannot be created. \n"
+            message += "Please run 'python imagenet/model_saver.py --device GPU' on a node with an available CUDA device."
+            logger.error(message)
+            sys.exit(1)
+
+def create_folder(exp_name, launcher): 
+    i = 0
+    while os.path.exists("results/"+exp_name+"/run" + str(i)):
+        i += 1
+    path2 = os.path.join(os.getcwd(), "results/"+exp_name+"/run" + str(i)) #autoincrement
+    os.makedirs(path2)
+    exp = Experiment(name="results/"+exp_name+"/run" + str(i), launcher=launcher)
+    exp.generate()
+    log_to_file(f"{exp.exp_path}/scaling-{get_date()}.log")
+    return exp
+
+def start_database(exp, node_feature, port, nodes, cpus, tpq, net_ifname, run_as_batch, hosts):
```

Comment:
My understanding was that `batch_args` was param that literally just took a dictionary of whatever and handed it to an SS `RunSettings` derivative making `batch_args` a superset of functionality of `node_feature`.

That said, it looks like the `batch_args` were only ever realistically used in the inference studies, so I don't see any problem making this change, but I wanted to make sure I was on the same page! We'll also need to remove the arg from the throughput and agg studies as well though

Reply:
Im gonna say that we change it since its more specific than batch_args? Since we can rotate between specifyinf db_node_feature and just node_feature, what do u think? Or maybe node_feature isnt more specific than batch_args?